cmd/coder/urls.go

```diff
@@ -35,47 +37,66 @@ var urlAccessLevel = map[string]string{
 
 func portIsValid(port string) bool {
 	p, err := strconv.ParseUint(port, 10, 16)
-	if p < 1 {
+	if p < 1 || p > math.MaxUint16 {
```

Comment:
It's not possible for this function to return a value over the provided bitsize anyways, so this check isn't doing much. See https://play.golang.org/p/8aTqC70r7Ou

I think this error handling here is a little backwards. We should be checking `err` before checking `p` since `p` will be 0 if `err != nil`. If we're checking that `p < 1` before `err` then we'll be returning the wrong error.

Reply:
OK. Duh, yes I see now I'd already called ParseUint() so it has to be in range :)